app/controllers/posts_controller.rb

```diff
@@ -7,5 +7,20 @@ def index
   def show
     @post = Post.find_by(id: params[:id])
     @comments = Comment.where(post_id: params[:id])
+    @user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
```

Comment:
[OPTIONAL] I think it's best to create a private params method because it is reusable.

```
    private 

    def post_params
      params.require(:post).permit(:title, :text)
    end
```

Then you can simply create a new post like this:

```
    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
       ...
      end
    end
```